tz/digest.go

```diff
@@ -122,3 +126,46 @@ func mulBitRightGeneric(c00, c10, c01, c11 *GF127, bit bool, tmp *GF127) {
 		*c11 = *tmp
 	}
 }
+
+// MarshalBinary implements [encoding.BinaryMarshaler].
+func (d *digest) MarshalBinary() ([]byte, error) {
+	var (
+		b = make([]byte, 0, Size)
+	)
+
+	for _, a := range d.x {
+		state, err := a.MarshalBinary()
+		if err != nil {
+			return nil, err
+		}
+
+		b = append(b, state...)
+	}
+
+	return b, nil
+}
+
+// UnmarshalBinary implements [encoding.BinaryUnmarshaler].
+func (d *digest) UnmarshalBinary(b []byte) error {
+	if len(b) != Size {
+		return errors.New("tz: invalid hash state size")
+	}
+
+	var (
+		start, end int
+	)
+
+	for i := 0; i < 4; i++ {
+		var g GF127
+		start = gf127.Size * i
+		end = start + gf127.Size
+
+		if err := g.UnmarshalBinary(b[start:end]); err != nil {
```

Comment:
Why not `d.x[i].UnmarshalBinary()`, btw?

Reply:
Removed extra var